### DISCUSSION TITLE
Help on updating to V4

### DISCUSSION THREAD

**Discussion by MatD1** [nextauthjs/next-auth#3099] [Help]
As per older posts I am new to messing with NextAuth, I use a Strapi backend for a lot of things and I love the usability of it. I also love NextAuth and its ease of use however I could not figure how to use Strapis built in auth system with NextAuth until they posted a guide on the Strapi blog [here](https://strapi.io/blog/user-authentication-in-next-js-with-strapi), however the guide uses v3 of NextAuth and I personally rather v4 and the various improvements it has. 

#### Problem
For the life of me I cannot figure out to convert the [...nextauth].js file to v4!

Any help or suggestions on the best way to do this would be amazing! 
~ thanks in advanced

[Github](https://github.com/ghoshnirmalya/nextjs-authentication-using-strapi-and-next-auth) repo from blog post

```javascript
// [...nextauth].js file v3

import NextAuth from "next-auth";
import Providers from "next-auth/providers";

const options = {
  providers: [
    Providers.Twitch({
      clientId: process.env.TWITCH_CLIENT_ID,
      clientSecret: process.env.TWITCH_SECRET,
    }),
  ],
  session: {
    jwt: true,
  },
  debug: true,
  callbacks: {
    session: async (session, user, token) => {
      session.jwt = user.jwt;
      session.id = user.id;
      return Promise.resolve(session);
    },
    jwt: async (token, user) => {
      const isSignIn = user ? true : false;

      if (isSignIn) {
        const response = await fetch(
          `${process.env.NEXT_PUBLIC_API_URL}/auth/${account.provider}/callback?access_token=${account?.accessToken}`
        );

        const data = await response.json();

        token.jwt = data.jwt;
        token.id = data.user.id;
      }

      return Promise.resolve(token);
    },
  },
};

const Auth = (req, res) => NextAuth(req, res, options);

export default Auth;
```

**Comment by balazsorban44** [comment]
https://next-auth.js.org/getting-started/upgrade-v4
